src/unity/toolkits/supervised_learning/supervised_learning.hpp

```diff
@@ -670,6 +693,15 @@ class EXPORT supervised_learning_model_base : public ml_model_base {
   variant_map_type api_evaluate(
       gl_sframe data, std::string missing_value_action, std::string metric);
 
+  /**
+   *  API interface through the unity server.
+   *
+   *  Evaluate the model and retun with predictions
+   */
+  // TODO: This function should be const
+  variant_map_type api_evaluate_with_predictions(
```

Comment:
I'm a bit confused as to why we have both `evaluate_with_predictions` and `api_evaluate_with_predictions` (in addition to being confused about why we have both in addition to `evaluate`). Which are intended to be used for which use cases? The API interfaces only seem subtly different.

Reply:
It's the wrapper around `evaluate_with_predictions` to use the `variant_type` api in Python. This probably should be refactored to use the new function registration mechanism instead of variant type.

Reply:
Notes to self:
- Removed `api_evaluate_with_predictions` into `evaluate_with_predictions`
- Refactoring `evaluate_with_predictions` into `evaluate`

Reply:
refactored